DOCSP-47550 Update the docker documentation

jwilson-mdb opened the pull request:
## DESCRIPTION
Adds note to the docker documentation clarifying that it is for local development only

**Note for Reviewer**
- The build log contains many `Page not included in any toctree...` errors
  - Unused pages will be removed in [DOCSP-47546](https://jira.mongodb.org/browse/DOCSP-47546)
  - Pages are being left in place for now to facilitate: 
      - content reorganization updates
      - creation of redirects before publishing feature branch 

## STAGING
https://deploy-preview-214--docs-relational-migrator.netlify.app/installation/install-with-docker/#install-with-docker

## JIRA
https://jira.mongodb.org/browse/DOCSP-47550

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

davidsoulmdb commented:
I would suggest dialling it down slightly?

For Evaluation Purposes
This Docker image is intended as an easy way to evaluate Relational Migrator. It has not been designed for resilience and is therefore not recommended for production workloads.

Current:
Warning
This deployment method is intended for evaluating Relational Migrator. It is not recommended for production workloads, and may not provide a resilient production environment.

jwilson-mdb commented:
> I would suggest dialling it down slightly?
> 
> For Evaluation Purposes This Docker image is intended as an easy way to evaluate Relational Migrator. It has not been designed for resilience and is therefore not recommended for production workloads.
> 
> Current: Warning This deployment method is intended for evaluating Relational Migrator. It is not recommended for production workloads, and may not provide a resilient production environment.

@davidsoulmdb Happy to dial this back a bit. I've:
- downgraded it from a `warning` to a `note` (no more red)
- simplified the message to just state its not recommend for prod
- added a link to the prod install guide
 
Let me know what you think, thanks!

davidsoulmdb commented:
Thanks, looks good

davidsoulmdb reviewed (approved):
LGTM, thanks for the edits
